pkg/core/pipeline/source/main.go

```diff
@@ -1,63 +1,36 @@
 package source
 
 import (
-	"fmt"
 	"os"
 	"strings"
 
 	"github.com/sirupsen/logrus"
 
-	"github.com/mitchellh/mapstructure"
+	"github.com/updatecli/updatecli/pkg/core/pipeline/resource"
 	"github.com/updatecli/updatecli/pkg/core/pipeline/scm"
 	"github.com/updatecli/updatecli/pkg/core/result"
-	"github.com/updatecli/updatecli/pkg/core/transformer"
-	"github.com/updatecli/updatecli/pkg/plugins/awsami"
-	"github.com/updatecli/updatecli/pkg/plugins/dockerdigest"
-	"github.com/updatecli/updatecli/pkg/plugins/file"
-	"github.com/updatecli/updatecli/pkg/plugins/github"
-	"github.com/updatecli/updatecli/pkg/plugins/githubrelease"
-	"github.com/updatecli/updatecli/pkg/plugins/gittag"
-	"github.com/updatecli/updatecli/pkg/plugins/helm"
-	"github.com/updatecli/updatecli/pkg/plugins/jenkins"
-	"github.com/updatecli/updatecli/pkg/plugins/maven"
-	"github.com/updatecli/updatecli/pkg/plugins/shell"
-	"github.com/updatecli/updatecli/pkg/plugins/version"
-	"github.com/updatecli/updatecli/pkg/plugins/yaml"
 )
 
 // Source defines how a value is retrieved from a specific source
 type Source struct {
-	Changelog string // Changelog hold the changelog description
-	Result    string // Result store the source result after a source run. This variable can't be set by an updatecli configuration
+	Changelog string // Changelog holds the changelog description
+	Result    string // Result stores the source result after a source run. This variable can't be set by an updatecli configuration
 	Output    string // Output contains the value retrieved from a source
 	Config    Config // Config defines a source specifications
 	Scm       *scm.ScmHandler
 }
 
 // Config struct defines a source configuration
 type Config struct {
-	DependsOn    []string                 `yaml:"depends_on"` // DependsOn specify dag dependencies between sources
-	Name         string                   // Name contains a source name
-	Kind         string                   // Kind defines a source kind
-	Prefix       string                   // Deprecated in favor of Transformers on 2021/01/3
-	Postfix      string                   // Deprecated in favor of Transformers on 2021/01/3
-	Transformers transformer.Transformers // Transformers defines the list of transformers to apply to a source Output
-	Replaces     Replacers                // Deprecated in favor of Transformers on 2021/01/3
-	Spec         interface{}
-	Scm          map[string]interface{} // Deprecated field on version [x.y.z]
-	SCMID        string                 `yaml:"scmID"` // SCMID references a uniq scm configuration
-}
-
-// Sourcer source is an interface to handle source spec
-type Sourcer interface {
-	Source(workingDir string) (string, error)
+	resource.ResourceConfig `yaml:",inline"`
+	// Deprecated in favor of Transformers on 2021/01/3
```

Comment:
What does it mean?

Reply:
It's the comment associated to the line below (e.g. the attrobite Replaces Repalcers).

If find really hard comment at the end of a line to read, but I realize that most of the code of updatecli does it so I'll revert (I did it by habit while experimenting and it made it to this PR...)

Reply:
(applied a modification to make it a single line).

Reply:
I don't understand the meaning and goal of this comment.
Why something one year old should be mentioned?

Reply:
I don't either, but the goal of this PR is not to cleanup so I don't see the point of changing it here.

However it's legit: https://github.com/updatecli/updatecli/issues/497 just opened an issue to track it :)

Reply:
Oh I see, I initially thought you've created and added this comment, and thus I was wondering why, or if that was a typo.

Reply:
No problem, I understood that afterwards, hence the issue because it's totally legit :) Thanks!